src/lib/navigation/configurable-navigation-mode.ts

```diff
@@ -0,0 +1,107 @@
+import {BaseNavigationMode} from './base-navigation-mode.interface';
+import {WizardComponent} from '../components/wizard.component';
+import {WizardCompletionStep} from '../util/wizard-completion-step.interface';
+
+export class ConfigurableNavigationMode extends BaseNavigationMode {
```

Comment:
This class could also use some comments to explain when it should be used.

Reply:
Comments added in 1888a68